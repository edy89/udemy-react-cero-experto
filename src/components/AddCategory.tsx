import { useState } from 'react';

interface AddCategoryProps {
    onNewCategory: (value: string) => void;
  }
// interface AddCategoryProps {
//     setCategories: React.Dispatch<React.SetStateAction<string[]>>; // Tipo de setCategories
//   }

// export const AddCategory: React.FC<AddCategoryProps> = ({setCategories}) => {
export const AddCategory: React.FC<AddCategoryProps> = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');
    const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    // const onInputChange = ({target}) => {
        setInputValue(event.target.value);
    };
    const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault(); // avoiding refresh page that is default configuration with form.

        const newInputValue = inputValue.trim()

        if (newInputValue.trim().length <= 1) return;

        onNewCategory(newInputValue);
        // setCategories((categories: string[]) => [inputValue, ...categories]);
        setInputValue('');
    };

  return (
    <form onSubmit={(event) => {onSubmit(event)}}>
        <input 
            type="text" 
            placeholder="Search gifs"
            value={inputValue}
            onChange={ (event) =>  {onInputChange(event)}}
            // onChange={onInputChange}
        />
    </form>
  )
}
