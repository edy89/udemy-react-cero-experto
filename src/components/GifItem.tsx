
interface CardPropsImageI {
    title: string;
    url: string;
}

export const GifItem: React.FC<CardPropsImageI> = ({title, url}) => {
  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}
