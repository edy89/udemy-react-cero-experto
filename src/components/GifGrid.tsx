import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

interface GifGridPropsI {
    category: string;
}

interface PropsImageI {
    id: string;
    title: string;
    url: string;
}


export const GifGrid: React.FC<GifGridPropsI> = ({category}) => {

  const {images, isLoading} = useFetchGifs(category); // This a custom hook!
    

  return (
    <>
        <h3>{category}</h3>
        {
          isLoading && (<h2>Cargando...</h2>)
        }
        <div className="card-grid">
          {
            images.map((image: PropsImageI) => (
              <GifItem 
                key={image.id}
                { ...image }
                // title={title}
                // url={url}
              />
            ))
            // images.map((image: GetGifsPropsImageI) => (
            //   <li key={image.id}>{image.title}</li>
            // ))
          }
        </div>
        <p>This is a parragraph</p>
    </>
  )
}
