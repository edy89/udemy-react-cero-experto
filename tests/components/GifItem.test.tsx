import {render, screen} from '@testing-library/react';
import {GifItem} from '../../src/components/GifItem';
import React from 'react';

describe('Testing <GifItem/>', () => {
    const title = 'Kakaroto';
    const url = ''
    
    test('should match with the snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('should show the image with the correct url and ALT', () => {
        render(<GifItem title={title} url={url}/>);
        // screen.debug();
        const {src, alt} = screen.getByRole('img') as HTMLImageElement;

        expect (src).toBe(url);  
        expect (alt).toBe(title);
    });

    test('should show the title in the component', () => {
        render(<GifItem title={title} url={url}/>);
        // screen.debug();
        expect(screen.getByText(title)).toBeTruthy();
    });
});