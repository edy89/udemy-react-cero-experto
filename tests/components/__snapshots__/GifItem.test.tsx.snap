// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing <GifItem/> > should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Kakaroto"
      src=""
    />
    <p>
      Kakaroto
    </p>
  </div>
</div>
`;
